src/CourseDashboard.cpp

```diff
@@ -1,4 +1,218 @@
+            }
+          
+    
+    }while(test==false);
+   //   this->panel(*it);
+    
+}
+
+void CourseDashboard::run()
+{
+    Trainer Lukasz ;
+    Lukasz.name = "Lukasz1";
+    Lukasz.password="Ziobron";
+    ListOfGoldFathers.push_back(Lukasz);
+    Lukasz.name = "Ola";
+    Lukasz.password="Halapacz";
+    ListOfGoldFathers.push_back(Lukasz);
+    Lukasz.name = "Lukasz3";
+    Lukasz.password="Ziobron";
+    ListOfGoldFathers.push_back(Lukasz);
+    this->login();
+ 
+}
+/*
+bool CourseDashboard::checklog(std::string name,std::string password)
+{
+    return true;
+}
+*/
+void CourseDashboard::panel(Trainer &T)
+{
+    int ch = 0;
+
+	do
+	{
+		std::system("clear");
+
+		std::cout<< "1 - show list of students" <<std::endl;
+		std::cout<< "2 - add student" << std::endl;
+		std::cout<< "3 - remove student" << std::endl;
+		std::cout<< "4 - view student" << std::endl;
+		std::cout<< "5 - modify student" << std::endl;
+		std::cout << "---------------------------------------------------------------" << std::endl;
+		std::cout<< "0 - exit" << std::endl;
+		std::cout << "---------------------------------------------------------------" << std::endl;
+
+		std::cin>>ch;
+
+		switch(ch)
+		{
+			case 1: 
+				{
+					this->showStudents();
+					break;
+				}
+			case 2:
+				{
+					this->addStudent();
+					break;
+				}
+			case 3:
+				{
+					int name;
+					std::cout<< "Who will be erase?" <<std::endl;
+					std::cin >> name;
+					this->removeStudent(name);
+					break;
+				}
+            case 4: 
+                {
+                    int num;
+                    std::string change;
+                    std::cout<<"Write number student to edit: ";
+                    std::cin>>num;
+                     std::cout<<"What do you want to change?";
+                     std::cin>>change;
+                     if(change=="name") 
+                     {
+                         std::cout<<"Write new name";
+                         std::cin>>change;
+                         listOfStudents[num].name=change;
+                         break;
+                     }
+                     if(change=="nick") 
+                     {
+                         std::cout<<"Write new nick";
+                         std::cin>>change;
+                         listOfStudents[num].nick=change;
+                         break;
+                     }
+                     if(change=="password") 
+                     {
+                         std::cout<<"Write new name";
+                         std::cin>>change;
+                         listOfStudents[num].password=change;
+                         break;
+                     }
+                    
+                }
+                    
+			default:
+				{
+					std::cout<< "Wrong selection, please pick one more time ;)" <<std::endl;
+				}
+		}
+		sleep(3);
+	}
+	while (ch != 0);
+
+	std::cout << "Bye!" <<std::endl;
```

Comment:
Generally, too long methods. Most of them requires input from the user. It makes them untestable.